### PR TITLE
Removing Mixer-specific proto hack that converts Struct => interface{}

### PR DESCRIPTION
**Pull request by ozevren**
Mixer code has a hack that uses map[string]interface{} instead of types.Struct for google.protobuf.Struct type. This hack makes the proto not-marshalable.

Upon further investigation, it looks like this hack is not used anymore. This PR converts the code to use the vanilla version of the proto.

### CODE REVIEW

File: mixer/pkg/runtime/config/ephemeral.go [hunk 1]
```diff
@@ -355,12 +355,12 @@ func (e *Ephemeral) processDynamicInstanceConfigs(ctx context.Context, templates
 		var params map[string]interface{}
 		var err error
 		if inst.Params != nil {
-			if _, ok := inst.Params.(map[string]interface{}); !ok {
+			if params, err = toDictionary(inst.Params); err != nil {
```
Comment (ayj) [inline mixer/pkg/runtime/config/ephemeral.go]: I'm assuming `processDynamicInstanceConfigs` is (in)directly called at config ingestion time in mixer and not in the data path.

Comment (ozevren) [reply mixer/pkg/runtime/config/ephemeral.go]: yes

**Comment by ozevren** [comment]
ping. I'd like to get this in before it rots. I had to bang on dep so many times to get it to do just the right thing. I'd hate to repeat that exercise again. :)

File: mixer/pkg/runtime/config/config_test.go [hunk 2]
```diff
@@ -2440,38 +2430,6 @@ Attributes:
 `,
 		wantErr: "",
 	},
-	{
-		Name: "add handler - bad param type",
-		Events1: []*store.Event{updateEvent("a1.adapter.default", &adapter_model.Info{
```
Comment (mandarjog) [inline mixer/pkg/runtime/config/config_test.go]: Are these tests replaced by something else, or just removed?

Comment (ozevren) [reply mixer/pkg/runtime/config/config_test.go]: The original code path for these is gone. That's why I removed it. However, there is still some similar (unlikely) downstream error paths, which I am not sure is captured in other tests, so I resurrected them to cover those.

**Pull request merged**